File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        //Use a variable, count, to track the number of drops (where nums[i] > nums[i+1]).
        int drop = 0;
        for(int i = 0; i < nums.length; i++){
            // Compare current with the next (circular comparison)
            if(nums[i] > nums[(i + 1) % nums.length]){
                drop++;
            }
        }
        return drop <= 1;
    }
}

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        l = len(nums)
        rotations = 0
        for i in range(l):
            if nums[i] > nums[(i+1)%l]:
                rotations +=1
            if rotations > 1:
                return False
        return True

        